### PR TITLE
[viessmann] Reduce logging level in case of errors and improve error handling

### PR DESCRIPTION
**Pull request by cweitkamp**
- Reduce logging level in case of errors and improve error handling

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/api/ViessmannApi.java [hunk 1]
```diff
@@ -319,4 +303,29 @@ private Properties setHeaders() throws ViessmannAuthException {
...
+                    logger.warn("ViError: {} | Resetting Limit at {}", viError.getMessage(),
+                            viError.getExtendedPayload().getLimitResetDateTime());
+                    bridgeHandler.updateBridgeStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            String.format("API Call limit reached. Reset at %s",
+                                    viError.getExtendedPayload().getLimitResetDateTime()));
+                    bridgeHandler.waitForApiCallLimitReset(viError.getExtendedPayload().getLimitReset());
+                    break;
+                case HttpStatus.BAD_GATEWAY_502:
+                    logger.debug("ViError: {} | Device not reachable", viError.getMessage());
+                    bridgeHandler.updateBridgeStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
```
Comment (rogrun) [inline bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/api/ViessmannApi.java]: Do we really want to set the bridge to offline if a device is unreachable? Shouldn't the device go offline and not the bridge?

**Review by rogrun — CHANGES_REQUESTED** [review]

**Comment by rogrun** [comment]
The rest LGTM

File: bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/api/ViessmannApi.java (hunk 1 above)
Comment (cweitkamp) [reply bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/api/ViessmannApi.java]: Good point. I'll check that.

Comment (cweitkamp) [reply bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/api/ViessmannApi.java]: For a proper status handling of thing handlers I submitted https://github.com/smarthomej/addons/pull/428 to improve the polling.

Comment (cweitkamp) [reply bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/api/ViessmannApi.java]: Done. See https://github.com/smarthomej/addons/pull/426/commits/69e576fd77447243e2f8b6dd5c7216bd1339b657.

**Review by rogrun — APPROVED** [review]
LTGM

**Review by J-N-K — APPROVED** [review]

**Pull request merged**